frontend/src/metabase/parameters/components/ParameterSettings/ParameterSettings.tsx

```diff
@@ -46,11 +46,35 @@ const ParameterSettings = ({
   onChangeSourceConfig,
   onRemoveParameter,
 }: ParameterSettingsProps): JSX.Element => {
+  const [internalValue, setInternalValue] = useState(parameter.name);
+
+  useLayoutEffect(() => {
+    setInternalValue(parameter.name);
+  }, [parameter.name]);
+
+  const handleChange = useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      setInternalValue(event.target.value);
+    },
+    [],
+  );
+
+  const labelInputError = useMemo(() => {
+    if (internalValue === "") {
+      return t`Required`;
+    }
+    return null;
+  }, [internalValue]);
+
   const handleNameChange = useCallback(
     (event: { target: HTMLInputElement }) => {
-      onChangeName(event.target.value);
+      if (labelInputError) {
```

Comment:
This indirection is actually useful because it means that the conditions for showing the error message on the TextInput
component are the same conditions for reverting the value here. Also there will be more error reasons in the future with
#34369, so this will become even more useful.

Reply:
> Also there will be more error reasons in the future with https://github.com/metabase/metabase/pull/34369, so this will become even more useful.

Fair enough :+1: